File: medplum-java/medplum-model/src/main/java/com/fhir/DeviceMetric.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableDeviceMetric.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface DeviceMetric {
  Optional<Narrative> text();

  Optional<Timing> measurementPeriod();

  Optional<DevicemetricColor> color();

  Optional<CodeableConcept> unit();

  Optional<id> id();

  Optional<DevicemetricOperationalstatus> operationalStatus();

  String resourceType();

  Optional<code> language();

  CodeableConcept type();

  Optional<List<DeviceMetric_Calibration>> calibration();

  Optional<List<ResourceList>> contained();

  Optional<List<Extension>> extension();

  Optional<Reference> source();

  Optional<Meta> meta();

  Optional<DevicemetricCategory> category();

  Optional<List<Extension>> modifierExtension();

  Optional<Reference> parent();

  Optional<uri> implicitRules();

  Optional<List<Identifier>> identifier();

  static ImmutableDeviceMetric.ResourceTypeBuildStage builder() {
    return ImmutableDeviceMetric.builder();
  }
}

File: medplum-typescript/src/model/fhir/DeviceMetric.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { DevicemetricCategory } from "./DevicemetricCategory";
import { DevicemetricColor } from "./DevicemetricColor";
import { DevicemetricOperationalstatus } from "./DevicemetricOperationalstatus";
import { DeviceMetric_Calibration } from "./DeviceMetric_Calibration";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { Timing } from "./Timing";
import { uri } from "./uri";

export interface DeviceMetric {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    type: CodeableConcept;
    unit: CodeableConcept | null | undefined;
    source: Reference | null | undefined;
    parent: Reference | null | undefined;
    operationalStatus: DevicemetricOperationalstatus | null | undefined;
    color: DevicemetricColor | null | undefined;
    category: DevicemetricCategory | null | undefined;
    measurementPeriod: Timing | null | undefined;
    calibration: DeviceMetric_Calibration[] | null | undefined;
}
